python/xoscar/collective/gloo/xoscar_pygloo.pyi

```diff
@@ -0,0 +1,191 @@
+    sendbuf: Optional[int] = None,
+    recvbuf: Optional[int] = None,
+    size: Optional[int] = None,
+    datatype: Optional[GlooDataType] = None,
+    reduceop: Optional[ReduceOp] = ReduceOp.SUM,
+    root: Optional[int] = 0,
+    tag: Optional[int] = 0,
+) -> None: ...
+def scatter(
+    context: Optional[Context] = None,
+    sendbuf: Optional[int] = None,
+    recvbuf: Optional[int] = None,
+    size: Optional[int] = None,
+    datatype: Optional[GlooDataType] = None,
+    root: Optional[int] = 0,
+    tag: Optional[int] = 0,
+) -> None: ...
+def gather(
+    context: Optional[Context] = None,
+    sendbuf: Optional[int] = None,
+    recvbuf: Optional[int] = None,
+    size: Optional[int] = None,
+    datatype: Optional[GlooDataType] = None,
+    root: Optional[int] = 0,
+    tag: Optional[int] = 0,
+) -> None: ...
+def send(
+    context: Optional[Context] = None,
+    sendbuf: Optional[int] = None,
+    size: Optional[int] = None,
+    datatype: Optional[GlooDataType] = None,
+    peer: Optional[int] = None,
+    tag: Optional[int] = 0,
+) -> None: ...
+def recv(
+    context: Optional[Context] = None,
+    recvbuf: Optional[int] = None,
+    size: Optional[int] = None,
+    datatype: Optional[GlooDataType] = None,
+    peer: Optional[int] = None,
+    tag: Optional[int] = 0,
+) -> None: ...
+def broadcast(
+    context: Optional[Context] = None,
+    sendbuf: Optional[int] = None,
+    recvbuf: Optional[int] = None,
+    size: Optional[int] = None,
+    datatype: Optional[GlooDataType] = None,
+    root: Optional[int] = 0,
+    tag: Optional[int] = 0,
+) -> None: ...
+def reduce_scatter(
+    context: Optional[Context] = None,
+    sendbuf: Optional[int] = None,
+    recvbuf: Optional[int] = None,
+    size: Optional[int] = None,
+    recvElems: Optional[List[int]] = None,
+    datatype: Optional[GlooDataType] = None,
+    reduceop: Optional[ReduceOp] = ReduceOp.SUM,
+) -> None: ...
+def barrier(context: Optional[Context] = None, tag: Optional[int] = 0) -> None: ...
+
+class rendezvous:
+    class Store:
+        def set(self, key: str, data: List[str]) -> None: ...
+        def get(self, key: str) -> str: ...
+
+    class FileStore(Store):
+        def __init__(self, path: str) -> None: ...
+
+    class HashStore(Store):
+        def __init__(self) -> None: ...
+
+    class PrefixStore(Store):
+        def __init__(self, prefix: str, store: rendezvous.Store) -> None: ...
+
+    class CustomStore(Store):
+        def __init__(self, real_store_py_object: object) -> None: ...
+        def delKeys(self, keys: List[str]) -> None: ...
+
+    class Context(xoscar_pygloo.Context):
+        def connectFullMesh(
+            self, store: rendezvous.Store, dev: xoscar_pygloo.transport.Device
+        ) -> None: ...
+
+class transport:
+    class uv:
+        pass
+
+    class tcp:
+        class Device(transport.Device):
+            def __init__(self, attr: transport.tcp.attr) -> None: ...
+
+        class Context(xoscar_pygloo.Context):
```

Comment:
Ditto.

Reply:
if I use `Context` instead of `xoscar_pygloo.Context`, there will be some errors in `pre-commit`